pages/agent/gcloud.md.erb

```diff
@@ -0,0 +1,202 @@
+           privileged: true
+         env:
+           - name: BUILDKITE_AGENT_TOKEN
+             valueFrom: {secretKeyRef: {name: buildkite-agent, key: token}}
+         volumeMounts:
+           - name: docker-binary
+             mountPath: /usr/bin/docker
+           - name: docker-socket
+             mountPath: /var/run/docker.sock
+     volumes:
+       - name: docker-binary
+         hostPath: {path: /usr/bin/docker}
+       - name: docker-socket
+         hostPath: {path: /var/run/docker.sock}
+# (press Control-D)
+
+deployment "buildkite-agent" created
+```
+
+Verify this started a [pod](https://kubernetes.io/docs/user-guide/pods/) running the agent:
+
+```shell
+$ kubectl get pods
+NAME                               READY     STATUS    RESTARTS   AGE
+buildkite-agent-3220167863-dtzcf   1/1       Running   0          30s
+```
+
+And verify that the agent has registered successfully in the logs:
+
+```shell
+$ kubectl logs buildkite-agent-3220167863-dtzcf
+
+  _           _ _     _ _    _ _                                _
+ | |         (_) |   | | |  (_) |                              | |
+ | |__  _   _ _| | __| | | ___| |_ ___    __ _  __ _  ___ _ __ | |_
+ | '_ \| | | | | |/ _` | |/ / | __/ _ \  / _` |/ _` |/ _ \ '_ \| __|
+ | |_) | |_| | | | (_| |   <| | ||  __/ | (_| | (_| |  __/ | | | |_
+ |_.__/ \__,_|_|_|\__,_|_|\_\_|\__\___|  \__,_|\__, |\___|_| |_|\__|
+                                                __/ |
+ http://buildkite.com/agent                    |___/
+2017-02-16 03:34:10 NOTICE Starting buildkite-agent v2.3.2 with PID: 5
+2017-02-16 03:34:10 NOTICE The agent source code can be found here: https://github.com/buildkite/agent
+2017-02-16 03:34:10 NOTICE For questions and support, email us at: hello@buildkite.com
+2017-02-16 03:34:10 INFO   Registering agent with Buildkite...
+2017-02-16 03:34:11 INFO   Successfully registered agent "buildkite-agent-3220167863-1cvr5" with meta-data []
+2017-02-16 03:34:11 INFO   Connecting to Buildkite...
+2017-02-16 03:34:11 INFO   Agent successfully connected
+2017-02-16 03:34:11 INFO   You can press Ctrl-C to stop the agent
+2017-02-16 03:34:11 INFO   Waiting for work...
+```
+
+You’re successfully running a Buildkite Agent!
+
+To run more than one agent you can increase `replicas`:
+
+```shell
+$ kubectl scale --replicas=5 deployment buildkite-agent
+deployment "buildkite-agent" scaled
+
+$ kubectl get pods
+NAME                               READY     STATUS    RESTARTS   AGE
+buildkite-agent-3220167863-0m269   1/1       Running   0          24s
+buildkite-agent-3220167863-0mtps   1/1       Running   0          24s
+buildkite-agent-3220167863-1cvr5   1/1       Running   0          3m
+buildkite-agent-3220167863-9psz9   1/1       Running   0          24s
+buildkite-agent-3220167863-m3nm2   1/1       Running   0          24s
+```
+
+To run builds from private repositories you may want to store [an SSH key for the agent](/docs/agent/ssh-keys) in a secret and map it into the containers:
```

Comment:
If they want to store an ssh key. There are multiple ways of accessing different private repositories with different providers, this is just a hint at what's possible for a popular option. I don't want to endorse SSH keys versus oauth tokens versus Google's repository offering with security contexts, just hint that there is an option for ssh keys.